airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/partition_enqueuer.py

```diff
@@ -34,6 +39,8 @@ def generate_partitions(self, stream: AbstractStream) -> None:
         """
         try:
             for partition in stream.generate_partitions():
+                while self._queue.qsize() >= self._max_size:
```

Comment:
this should have a max attempt

Reply:
Can we explain why this is needed on top of the priority? It's not so obvious why both would be needed